node/src/components/consensus/era_supervisor.rs

```diff
@@ -172,10 +182,17 @@ where
             stop_for_upgrade: false,
             next_executed_height: 0,
             is_initialized: false,
+            enqueued_events: Default::default(),
         };
 
+        let era_ids: Vec<EraId> = era_supervisor
+            .iter_past(current_era, bonded_eras * 3)
```

Comment:
With https://github.com/CasperLabs/casper-node/pull/979 merged, we won't need that many eras?

Reply:
I'm not sure actually.
Even with #979, we still need to initialize `2 * bonded_eras` eras, and for that we need `3 * bonded_eras` switch blocks.
Although maybe for the older half of those eras, which only need to validate evidence, we don't need the cross-era slashing information and the booking block hash, because we don't accept any units or blocks anymore? So we could initialize them with a wrong seed and wrong cross-era slashing information…
But I'd prefer to do that cleanly at some point: Actually make `HighwayProtocol` an enum, and make it so that the `EvidenceOnly` variant doesn't require those old switch blocks for initialization anymore. And expose that information in `ConsensusProtocol` in a way that's not Highway-specific.

Reply:
I'm not sure. I believe we might still try to look that far back. Also, note that this is the number of switch block we're trying to read, the actual number of eras will be `* 2`, not `* 3`.
@afck Do you think something should be changed here?

Reply:
In this PR I'd definitely keep it `* 3`.